Core/src/Detector/CylindricalContainerBuilder.cpp

```diff
@@ -0,0 +1,149 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Detector/CylindricalContainerBuilder.hpp"
+
+#include "Acts/Detector/DetectorComponents.hpp"
+#include "Acts/Detector/detail/CylindricalDetectorHelper.hpp"
+#include "Acts/Utilities/BinningData.hpp"
+
+#include <stdexcept>
+
+namespace {
+
+/// @brief Helper method to connect/wrap volumes or containers
+///
+/// @tparam object_collection either a vector of volumes or containers
+/// @param gctx The geometry context of the this call
+/// @param objects The object vector
+/// @param binnning the chosen binning
+/// @param logLevel the logging output level
+///
+/// @note no checking on consistency is done as the caller container builder
+/// is already checked at construction
+///
+/// @return a newly built container
+template <typename object_collection>
+Acts::Experimental::DetectorComponent::Container connect(
+    const Acts::GeometryContext& gctx, object_collection& objects,
+    const std::vector<Acts::BinningValue>& binning,
+    Acts::Logging::Level logLevel) {
+  // Return container object
+  Acts::Experimental::DetectorComponent::Container rContainer;
+  if (binning.size() == 1u) {
+    Acts::BinningValue bv = binning.front();
+    // 1-dimensional binning options
+    switch (bv) {
+      case Acts::binR: {
+        rContainer =
+            Acts::Experimental::detail::CylindricalDetectorHelper::connectInR(
+                gctx, objects, {}, logLevel);
+      } break;
+      case Acts::binZ: {
+        rContainer =
+            Acts::Experimental::detail::CylindricalDetectorHelper::connectInZ(
+                gctx, objects, {}, logLevel);
+      } break;
+      case Acts::binPhi: {
+        rContainer =
+            Acts::Experimental::detail::CylindricalDetectorHelper::connectInPhi(
+                gctx, objects, {}, logLevel);
+      } break;
+      default:
+        break;
+    }
+  } else if (binning ==
+                 std::vector<Acts::BinningValue>{Acts::binZ, Acts::binR} and
+             objects.size() == 2u) {
+    rContainer =
+        Acts::Experimental::detail::CylindricalDetectorHelper::wrapInZR(
+            gctx, objects, logLevel);
+  }
+  return rContainer;
```

Comment:
Do you assume `binning` being only 1- and 2-dim? Shall we add a warning line in case the size is not expected?

Reply:
See below.